File: ServerStatusNotifierModel/src/main/java/com/github/grzegorzekkk/serverstatusnotifier/serverstatusnotifiermodel/ServerDetails.java
package com.github.grzegorzekkk.serverstatusnotifier.serverstatusnotifiermodel;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serializable;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class ServerDetails implements Serializable {
    public static final String SERIAL_NAME = "SERVER_DETAILS";

    private SrvConnDetails connDetails;
    private String serverName;
    private boolean isOnline;
    private int playersCount;
    private int playersMax;
    private String serverVersion;

    public ServerDetails(SrvConnDetails connDetailsArg, boolean isOnlineArg) {
        connDetails = connDetailsArg;
        isOnline = isOnlineArg;
    }
}

File: ServerStatusNotifier/app/src/main/java/com/github/grzegorzekkk/serverstatusnotifier/serverdetails/model/ServerDetails.kt
package com.github.grzegorzekkk.serverstatusnotifier.serverdetails.model

import android.os.Parcel
import android.os.Parcelable
import com.github.grzegorzekkk.serverstatusnotifier.database.SrvConnDetails
import com.github.grzegorzekkk.serverstatusnotifier.serverslist.model.ServerStatus

data class ServerDetails(val serverStatus: ServerStatus, var srvConnDetails: SrvConnDetails, val playersCount: Int) : Parcelable {
    constructor(parcel: Parcel) : this(
            parcel.readParcelable(ServerStatus::class.java.classLoader),
            parcel.readParcelable(SrvConnDetails::class.java.classLoader),
            parcel.readInt())

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeParcelable(serverStatus, flags)
        parcel.writeParcelable(srvConnDetails, flags)
        parcel.writeInt(playersCount)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<ServerDetails> {
        const val SERIAL_NAME = "serverDetails"

        override fun createFromParcel(parcel: Parcel): ServerDetails {
            return ServerDetails(parcel)
        }

        override fun newArray(size: Int): Array<ServerDetails?> {
            return arrayOfNulls(size)
        }
    }
}
